src/opentimelineio/imageSequenceReference.cpp

```diff
@@ -0,0 +1,136 @@
+    } 
+
+    std::string
+    ImageSequenceReference::target_url_for_image_number(int const image_number, ErrorStatus* error_status) const {
+        if (image_number >= this->number_of_images_in_sequence()) {
+            *error_status = ErrorStatus(ErrorStatus::ILLEGAL_INDEX);
+            return std::string();
+        }
+        const int file_image_num = _start_value + (image_number * _value_step);
+        const bool is_negative = (file_image_num < 0);
+        std::string image_num_string = std::to_string(abs(file_image_num));
+        std::string zero_pad = std::string();
+        if (image_num_string.length() <  _image_number_zero_padding) {
+            zero_pad = std::string(_image_number_zero_padding - image_num_string.length(), '0');
+        }
+
+        std::string sign = std::string();
+        if (is_negative) {
+            sign = "-";
+        }
+
+        std::string out_string = _target_url_base + _name_prefix + sign + zero_pad + image_num_string + _name_suffix;
+        *error_status = ErrorStatus(ErrorStatus::OK);
+        return out_string;
+    }
+
+    RationalTime
+    ImageSequenceReference::presentation_time_for_image_number(int const image_number, ErrorStatus* error_status) const {
+        if (image_number >= this->number_of_images_in_sequence()) {
+            *error_status = ErrorStatus(ErrorStatus::ILLEGAL_INDEX);
+            return RationalTime();
+        }
+
+        auto first_frame_time = this->available_range().value().start_time();
+        auto time_multiplier = TimeTransform(first_frame_time, image_number, -1);
+        return time_multiplier.applied_to(frame_duration());
+    }
+
+    bool ImageSequenceReference::read_from(Reader& reader) {
+        auto result = reader.read("target_url_base", &_target_url_base) &&
+                reader.read("name_prefix", &_name_prefix) &&
+                reader.read("name_suffix", &_name_suffix) &&
+                reader.read("start_value", &_start_value) &&
+                reader.read("value_step", &_value_step) &&
+                reader.read("rate", &_rate) &&
+                reader.read("image_number_zero_padding", &_image_number_zero_padding);
+
+                std::string missing_frame_policy_value;
+                result && reader.read("missing_frame_policy", &missing_frame_policy_value);
+                if (!result) {
+                    return result;
+                } 
+
+                if (missing_frame_policy_value == "error") {
+                    _missing_frame_policy = MissingFramePolicy::error;
+                }
+                else if (missing_frame_policy_value == "black") {
+                    _missing_frame_policy = MissingFramePolicy::black;
+                }
+                else if (missing_frame_policy_value == "hold") {
+                    _missing_frame_policy = MissingFramePolicy::hold;
+                }
+                else {
+                    // TODO: This fails silently, what should we do?
```

Comment:
I propose default to "error" if the key is missing, or raise a malformed file warning or error otherwise.

Reply:
A bit late to the party here.. I agree with @meshula that it should default to one of the policies and "error" sounds like a good one.